triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/AbstractFuncTests.scala

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.engine.trigger.test
+
+import akka.stream.scaladsl.{Flow}
+import com.digitalasset.daml.lf.data.Ref._
+import com.digitalasset.daml.lf.speedy.SExpr
+import com.digitalasset.daml.lf.speedy.SExpr._
+import com.digitalasset.daml.lf.speedy.SValue._
+import com.digitalasset.daml.lf.value.Value.AbsoluteContractId
+import com.digitalasset.ledger.api.testing.utils.{SuiteResourceManagementAroundAll}
+import com.digitalasset.ledger.api.v1.commands._
+import com.digitalasset.ledger.api.v1.commands.CreateCommand
+import com.digitalasset.ledger.api.v1.{value => LedgerApi}
+import com.digitalasset.platform.services.time.TimeProviderType
+import org.scalatest._
+import scalaz.syntax.traverse._
+
+import com.digitalasset.daml.lf.engine.trigger.TriggerMsg
+
+abstract class AbstractFuncTests
+    extends AsyncWordSpec
+    with AbstractTriggerTest
+    with Matchers
+    with SuiteResourceManagementAroundAll
+    with TryValues {
+  self: Suite =>
+
+  this.getClass.getSimpleName can {
+    "AcsTests" should {
+      val assetId = LedgerApi.Identifier(packageId, "ACS", "Asset")
+      val assetMirrorId = LedgerApi.Identifier(packageId, "ACS", "AssetMirror")
+      def asset(party: String): CreateCommand =
+        CreateCommand(
+          templateId = Some(assetId),
+          createArguments = Some(
+            LedgerApi.Record(fields =
+              Seq(LedgerApi.RecordField("issuer", Some(LedgerApi.Value().withParty(party)))))))
```

Comment:
I currently duplicate these 7 lines for the tls and auth tests. Couldn’t quite find a sensible location  to put them an dfor now this doesn’t seem like an issue.